### PR TITLE
Release 2.9.0

### PR DESCRIPTION
**Pull request by SachinSetiya**
fca9078d Exponentially backoff when out of background workers
5c8b0b52 Adjust ts_get_node_name for PG15
0d7da622 Adjust reorder code for PG15
9eef2e70 Don't error when compression metadata is missing
fe8d823b Add compat macros for PG15 signature changes
83eb635e Fix coverity warning about potential overflow
0d5049cf Fix max backoff on MAX_CONSECUTIVE_FAILURES
b34b91f1 Add timezone support to time_bucket_gapfill
d65cad94 Reorganize gapfill header files
6de97951 Fix compression_chunk_size primary key
f27e6273 Fix chunk exclusion for space partitions in SELECT FOR UPDATE queries
a26a5974 Improve space constraint exclusion datatype handling
4e47302c Speed up chunk search by restriction clauses
b869f91e Show warnings during create_hypertable().
02ad4f6b Change parameter names of cagg_migrate procedure
0144c75b Remove flaky test from bgw_db_scheduler
7aadf133 Fix some clang compile warnings
042735f1 Fix vacuum_set_xid_limits_compat() macro
7508d666 Fix gapfill function signature
217f5146 Fix continuous aggregate migrate check constraint
893faf8a Fix Continuous Aggregate migration policies
9d0d159a Port perl tests to support PG15
f6dd55a1 Hypertable FK reference to partitioned table
9c819882 Increase memory usage for compression jobs
cc110a33 Move ANALYZE after heap scan during compression
94018793 Fix segfault when INNER JOINing hypertables
9bd772de Add interface for troubleshooting job failures
eac3132b Migrate CI summary slack bot to eng-database
fa9d07f6 Add link to community forum to issue template chooser
ea5038f2 Add connection cache invalidation ignore logic
f1c6fd97 Continue compressing other chunks after an error
33e4e554 Ensure that internal users don't trigger workflows
8a5e59b0 Fix flaky bgw_custom test
b259191d Add COSTS OFF to make test more stable
8cda0e17 Extend the now() optimization to also apply to CURRENT_TIMESTAMP
a76f76f4 Improve size utils functions and views performance
d2f0c4ed Fix update script handling of bgw_job_stat
d602c1fe Fix typo in homebrew workflow
45a8c0b5 Improve formatting when printing coredump information
ea1014fa Fix flaky bgw_custom test
e0bbd404 Fix missing upgrade/downgrade tests DDL validation
efbd8a94 Increase timeout to wait for cluster start
2defb2b0 Improve job_crash_log test
7600896a Fix libpq disconnect for PG15
30596c0c Batch rows on access node for distributed COPY
7758f595 Update .clang-format for version 14
fbe4d3c1 Fix a warning with clang 14
459c3657 Ignore clang-format-14 changes in git blame
cc7ea8ef Fix ereport call in dist_copy for PG 12.0
e33bd897 Adjust TAP tests permissions
8f5698f4 Show information about OOM killer in CI
38878bee Fix segementation fault during INSERT into compressed hypertable.
066bcbed Rename row-by-row fetcher to COPY fetcher
d5c25e89 Adjust partition pruning to support PG15
0e32656b Support for PG15.
bde337e9 Fix the flaky pg_dump test
043bd55c Miss segmentby compression option in CAGGs
8950abe0 Bump github action versions
54ed0d5c Introduce fixed schedules for background jobs
702ac53c Bump codecov github action version
080011d7 Speed up the dist_copy tests
d218715d Fix tests which fail on PG15
f55aaf06 Add hook for ssl options
276d3a33 Add macro to assert or error
05ba1cf2 Add clang-tidy warning readability-suspicious-call-argument
73c3d02e Enable clang-tidy warning readability-misleading-indentation
f862212c Add clang-tidy warning readability-inconsistent-declaration-parameter-name
40a6c4cf Fix unused sort in dimension partition lookup
e555eea9 Fix performance regressions in the copy code
23c01c44 Remove an unused function in the copy code
4b054025 Add health check function
84b2fef6 Fix GitHub output action
25628e03 Use non-refcounted tupdesc for multi-insert buffers
39c99219 Fix flaky copy_memory_usage tests
06f2e57c Fix ereport call in job schedule for PG 12.0
fabb01fd Add telemetry for fixed schedule jobs
f0554588 Add a Coccinelle test for PG 12.3+ ereport syntax
cef8c462 Fix postgres version assert in telemetry
498b8af2 Block multi-statement DDL command in one query
da9af2c0 Do not cache the classify_relation result
2e7d7ee9 Remove useless message from tsl_cagg_try_repair
864da20c Build on Ubuntu 22.04
313845a8 Enable -Wextra
9b157d54 Don't use docker for clang-format
8d1e165d Refactor Continuous Aggregate catalog code
d8e892a6 Save postgres logs on Windows in CI
85f5efdc Fix -Wsign-compare with PG 15
dfbf030a Remove no-activity label on issue activity
c48b1231 Allow foreign tables in hypetable modify path
5b2d9d5a Enable -Wnewline-eof
e08e0a59 Add hook for chunk creation
d51fefb7 Add the Flaky Check
3d30f07b Swap lookup order for clang-format
20cdd9ca Fix bgw_db_scheduler_fixed flakyness
1cc8c15c Do not clobber the baserel cache on UDF error
840f144e Enable and fix -Wclobbered
c06b6476 pg_dump on PG15 does not log messages with log level set to PG_LOG_INFO.
8d1755bd Fix a typo in process_compressed_data_out()
2475c1b9 Roll up uncompressed chunks into compressed ones
08791cad Disable llvm on macos
7dd45cf3 Fix failure resuming a CAgg migration
f1535660 Honor usage of OidIsValid() macro
c54cf3ea Add job execution statistics to telemetry
1847f64a Fix sanitizer builds with -Wclobbered
6c73b61b Fix orphan jobs after CAgg migration
12745c88 Fix error: variable not found in subplan target list on PG15
f289ef88 Remove unused function ts_chunk_is_uncompressed_or_unordered
5b0bff38 Improve InvalidOid coccinelle check
3059290b Add new chunk state CHUNK_STATUS_COMPRESSED_PARTIAL
b9557655 Add printout for multiple jobs with same job_id
3a9688cc Extra Result node on top of CustomScan on PG15
2a644506 Add new tests to gitignore list
bfef3173 Refactor CAgg migration code to use job API
9744b4f3 Remove BitmapScan support in DecompressChunk
1f807153 Check for trailing whitespace
e4ba2bcf Remove debian 9 from packages tests.
6ad28248 Change the flaky check to use output files
4f9eef32 Print correct variables in downgrade test script
fe6731ce Fix compress_segmentby in isolation tests
f1321489 Add function to alter data nodes
252cefb5 Upload test results into the database
6ae19263 Fix CAgg migration with timestamp without timezone
9e276c58 Revert "Upload test results into the database"
e2b7c76c Disable MERGE when using hypertables
141e114c Fix race in bgw_db_scheduler_fixed
feb09c54 Rebuild cached PG daily and on config changes
0360812e Simplify llvm configuration for linux/macos builds
0d30155b Upload test results into a database
9964ba8b Remove accidental debug output
121631c7 Support parameterized data node scans in joins
b085833f Print errors in release builds for jobs
8afdddc2 Deprecate continuous aggregates with old format
33531212 Disable dist_move_chunk test in PG15
2f237e6e Add Enterprise Linux 9 packages to RPM package test
87756bcf Bump postgres versions used in CI
8b6eb902 Check for interrupts in gapfill main loop
51e5f319 Update compress chunk interval on compressed data
f3a3da78 Take advisory lock for job tuple
676d1fb1 Fix const null clauses in runtime chunk exclusion
7e4ebd13 Escape the quotes in gdb command
1b65297f Fix memory leak with INSERT into compressed hypertable
839e42dd Use async API to drop database from delete_data_node
1e3200be USE C function for time_bucket() offset
bfa641a8 INSERT .. SELECT on distributed hypertable fails on PG15
b9ca06d6 Move freeze/unfreeze chunk to tsl
7c32ceb0 Fix perl test import in PG15
380464df Perform frozen chunk status check via trigger
fd114797 Speed up get_or_add_baserel_from_cache operation
3749953e Hierarchical Continuous Aggregates
b1742969 Add SQL test files to trailing whitespace CI check
a4356f34 Remove trailing whitespaces from test code
3b5653e4 Ignore trailing whitespaces changes in git blame
89cede81 Fix PG15 specific tests.
a5b8c9b0 Fix caggs on caggs tests on PG15
fd84bf42 Use Ensure in get_or_add_baserel_from_cache
48d9733f Add telemetry for caggs on top of caggs
639a5018 Change time of scheduled CI run
7bc6e56c Fix plan_hashagg test failure in PG15
e84a6e2e Remove the refresh step from CAgg migration
35fa8910 Add missing gitignore entry
40297f18 Fix TRUNCATE on hierarchical caggs
26db8666 Fix GITHUB_OUTPUT on Windows
bdae647f Add i386 check results to database
5813173e Introduce drop_stale_chunks() function
7bfd28a0 Fix dist_fetcher_type test on PG15
826dcd27 Ensure nodes availability using dist restore point
26e3be14 Test dist caggs with an unavailable data node
c92e29ba Fix DML HA in multi-node
4a30e596 Fix flaky bgw_db_scheduler_fixed test
ce778faa Updating scheduled run
3b94b996 Use custom node to block frozen chunk modifications
83b13cf6 Use packaged postgres for sqlsmith and coverity CI
35c91204 Add Hierarchical Continuous Aggregates validations
2d0087a0 Fix segfault in cagg creation

### CODE REVIEW

**Comment by SachinSetiya** [comment]
I have removed this commit from the main branch. 
<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>

2f739bb3 | Post-release fixes for 2.8.1
-- | --



</body>
</html>

**Comment by SachinSetiya** [comment]
Ts_bench results
https://grafana.ops.savannah-dev.timescale.com/d/NdmLnOk4z/compare-benchmark-runs?orgId=1&var-run1=1669&var-run2=1671&var-postgres=14&var-branch=All&var-threshold=0

**Comment by SachinSetiya** [comment]
pre-release test https://github.com/timescale/timescaledb/actions?query=branch%3Aprerelease_test

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,41 +4,65 @@
...
-specifying the timezone to bucket for. Unfortunately this introduces
-an ambiguity with some previous call variations when an untyped
-start/finish argument is passed to time_bucket_gapfill. Some queries
-might need to be adjusted and either explicitly name the positional
-argument or resolve the type ambiguity by casting to the intended type.
+This release also includes several bug fixes.
 
 **Features**
+* #4476 Batch rows on access node for distributed COPY
+* #4567 Exponentially backoff when out of background workers
```
Comment (akuzm) [inline CHANGELOG.md:21]: These two don't look like feature, it's an implementation detail. But anyway, I see that this is a copy of changelog from main so maybe we can leave it as is.

**Review by akuzm — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (SachinSetiya) [reply CHANGELOG.md:21]: Okay

File: sql/updates/2.8.1--2.9.0.sql [hunk 2]
```diff
@@ -0,0 +1,496 @@
+-- gapfill with timezone support
```
[the remark this answers is not in the document]
Comment (SachinSetiya) [reply sql/updates/2.8.1--2.9.0.sql:1]: Right

File: tsl/src/compression/compression.c [hunk 3]
```diff
@@ -623,7 +623,6 @@ index_scan_sequence_number(Relation table_rel, Oid index_oid, ScanKeyData *scank
 	int32 result = 0;
 	bool is_null;
 	Relation index_rel = index_open(index_oid, AccessShareLock);
-	RelationInitIndexAccessInfo(index_rel);
```
Comment (nikkhils) [inline tsl/src/compression/compression.c:626]: Is this a relevant change?

Comment (SachinSetiya) [reply tsl/src/compression/compression.c:626]: Yes, it is pull request in main, I wanted to check if it helps with sanitizer fails, As soon as it is pushed into main, I will cherrypick it into 2.9.x

**Review by svenklemm — CHANGES_REQUESTED** [review]
This needs some additional fixups now since the release PR was merged to main already while the actual release was not ready yet.
